src/semantic/type_checker.rs

```diff
@@ -0,0 +1,161 @@
+#![allow(unused_variables)]
+use lexical::types::Types;
+use syntax::expr::ExprWrapper;
+use syntax::expr::Expr::*;
+use semantic::analyzer_trait::ASTAnalyzer;
+
+pub struct TypeChecker;
+
+impl TypeChecker {
+    pub fn new() -> TypeChecker {
+        TypeChecker
+    }
+
+    fn cmp_lhs_rhs(lhs_str: String, rhs_str: String) -> Option<String> {
```

Comment:
jk, you do return the types, ugh. yea that's why i don't necessarily like not including `return`. the difference between returning the value and not, is a lack of an `;`